src/daft-functions/src/list/unique.rs

```diff
@@ -0,0 +1,160 @@
+use std::{any::Any, sync::Arc};
+
+use arrow2::{compute::cast::cast, datatypes::DataType as ArrowDataType, offset::OffsetsBuffer};
+use common_error::{DaftError, DaftResult};
+use daft_core::{
+    array::{growable::make_growable, ListArray},
+    datatypes::{DataType, Field},
+    prelude::Schema,
+    series::{IntoSeries, Series},
+};
+use daft_dsl::{
+    functions::{ScalarFunction, ScalarUDF},
+    ExprRef,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct ListUnique {
+    ignore_nulls: bool,
+}
+
+#[typetag::serde]
+impl ScalarUDF for ListUnique {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn name(&self) -> &'static str {
+        "list_unique"
+    }
+
+    fn to_field(&self, inputs: &[ExprRef], schema: &Schema) -> DaftResult<Field> {
+        match inputs {
+            [input] => {
+                let field = input.to_field(schema)?;
+                match field.dtype {
+                    DataType::List(inner_type) => {
+                        Ok(Field::new(field.name, DataType::List(inner_type)))
+                    }
+                    DataType::FixedSizeList(inner_type, _) => {
+                        Ok(Field::new(field.name, DataType::List(inner_type)))
+                    }
+                    _ => Err(DaftError::TypeError(format!(
+                        "Expected list input, got {}",
+                        field.dtype
+                    ))),
+                }
+            }
+            _ => Err(DaftError::SchemaMismatch(format!(
+                "Expected 1 input arg, got {}",
+                inputs.len()
+            ))),
+        }
+    }
+
+    fn evaluate(&self, inputs: &[Series]) -> DaftResult<Series> {
+        match inputs {
+            [input] => {
+                let input = if let DataType::FixedSizeList(inner_type, _) = input.data_type() {
+                    let fixed_list = input.fixed_size_list()?;
+                    let arrow_array = fixed_list.to_arrow();
+                    let list_type = ArrowDataType::List(Box::new(arrow2::datatypes::Field::new(
+                        "item",
+                        inner_type.as_ref().to_arrow()?,
+                        true,
+                    )));
+                    let list_array = cast(arrow_array.as_ref(), &list_type, Default::default())?;
+                    Series::try_from_field_and_arrow_array(
+                        Field::new(input.name(), DataType::List(inner_type.clone())),
+                        list_array,
+                    )?
```

Comment:
I believe you can straight up cast a `FixedSizeList` type series into a `List` type instead of implementing this